src/main/java/network/brightspots/rcv/RawContestConfig.java

```diff
@@ -96,162 +100,233 @@ public static class CvrSource {
-      this.precinctColumnIndex = precinctColumnIndex;
+      this.precinctColumnIndex.set(precinctColumnIndex);
     }
 
     public String getOvervoteDelimiter() {
-      return overvoteDelimiter;
+      return overvoteDelimiter.get();
     }
 
     public void setOvervoteDelimiter(String overvoteDelimiter) {
-      this.overvoteDelimiter = overvoteDelimiter;
+      this.overvoteDelimiter.set(overvoteDelimiter);
     }
 
     public String getProvider() {
-      return provider;
+      return provider.get();
     }
 
-    public void setProvider(String provider) {
-      this.provider = provider;
+    /**
+     * Set the provider by either its GUI label or its internal label.
+     * Supporting both means we don't have to translate between internal and GUI labels
+     * each time we serialize/deserialize in the table.
+     *
+     * @param providerString The provider's internal or GUI label.
+     */
+    public void setProvider(String providerString) {
+      // First, try to get the provider by its public name
+      ContestConfig.Provider provider = ContestConfig.Provider.getByGuiLabel(providerString);
+
+      // If that fails, try its internal name
+      if (provider == ContestConfig.Provider.PROVIDER_UNKNOWN) {
+        provider = ContestConfig.Provider.getByInternalLabel(providerString);
+      }
+
+      this.provider.set(provider.getInternalLabel());
     }
 
     public String getOvervoteLabel() {
-      return overvoteLabel;
+      return overvoteLabel.get();
     }
 
     public void setOvervoteLabel(String overvoteLabel) {
-      this.overvoteLabel = overvoteLabel;
+      this.overvoteLabel.set(overvoteLabel);
     }
 
     public String getUndervoteLabel() {
-      return undervoteLabel;
+      return undervoteLabel.get();
     }
 
     public void setUndervoteLabel(String undervoteLabel) {
-      this.undervoteLabel = undervoteLabel;
+      this.undervoteLabel.set(undervoteLabel);
     }
 
     public String getUndeclaredWriteInLabel() {
-      return undeclaredWriteInLabel;
+      return undeclaredWriteInLabel.get();
     }
 
     public void setUndeclaredWriteInLabel(String undeclaredWriteInLabel) {
-      this.undeclaredWriteInLabel = undeclaredWriteInLabel;
+      this.undeclaredWriteInLabel.set(undeclaredWriteInLabel);
     }
 
-    public boolean isTreatBlankAsUndeclaredWriteIn() {
-      return treatBlankAsUndeclaredWriteIn;
+    public boolean getTreatBlankAsUndeclaredWriteIn() {
+      return treatBlankAsUndeclaredWriteIn.get();
     }
 
-    public void setTreatBlankAsUndeclaredWriteIn(boolean treatBlankAsUndeclaredWriteIn) {
-      this.treatBlankAsUndeclaredWriteIn = treatBlankAsUndeclaredWriteIn;
+    public void setTreatBlankAsUndeclaredWriteIn(Boolean treatBlankAsUndeclaredWriteIn) {
+      this.treatBlankAsUndeclaredWriteIn.set(treatBlankAsUndeclaredWriteIn);
+    }
+
+    /**
+     * The following properties are marked as unused by an editor, but
```

Comment:
A little confused by this comment. What is "an editor" referring to?

Reply:
IntelliJ and vim -- an IDE with some code analysis. I can replace with "IDE"

Reply:
Ahh gotcha, yeah, that would make more sense. "The following properties might be flagged as unused by an IDE, but..."